src/main/java/org/folio/inventory/ConsortiumInstanceSharingConsumerVerticle.java

```diff
@@ -0,0 +1,95 @@
+package org.folio.inventory;
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.inventory.consortium.consumers.ConsortiumInstanceSharingHandler;
+import org.folio.inventory.consortium.model.ConsortiumEvenType;
+import org.folio.inventory.dataimport.util.ConsumerWrapperUtil;
+import org.folio.inventory.storage.Storage;
+import org.folio.kafka.AsyncRecordHandler;
+import org.folio.kafka.GlobalLoadSensor;
+import org.folio.kafka.KafkaConfig;
+import org.folio.kafka.KafkaConsumerWrapper;
+import org.folio.kafka.KafkaTopicNameHelper;
+import org.folio.kafka.SubscriptionDefinition;
+import org.folio.processing.events.EventManager;
+
+import static java.lang.String.format;
+import static org.folio.inventory.dataimport.util.KafkaConfigConstants.KAFKA_ENV;
+import static org.folio.inventory.dataimport.util.KafkaConfigConstants.KAFKA_HOST;
+import static org.folio.inventory.dataimport.util.KafkaConfigConstants.KAFKA_MAX_REQUEST_SIZE;
+import static org.folio.inventory.dataimport.util.KafkaConfigConstants.KAFKA_PORT;
+import static org.folio.inventory.dataimport.util.KafkaConfigConstants.KAFKA_REPLICATION_FACTOR;
+import static org.folio.inventory.dataimport.util.KafkaConfigConstants.OKAPI_URL;
+
+public class ConsortiumInstanceSharingConsumerVerticle extends AbstractVerticle {
+
+  private static final Logger LOGGER = LogManager.getLogger(DataImportConsumerVerticle.class);
+
+  private final int loadLimit = getLoadLimit();
+
+  private KafkaConsumerWrapper<String, String> consumer;
+
+  private final int maxDistributionNumber = getMaxDistributionNumber();
+
+  @Override
+  public void start(Promise<Void> startPromise) {
+    JsonObject config = vertx.getOrCreateContext().config();
+    KafkaConfig kafkaConfig = KafkaConfig.builder()
+      .envId(config.getString(KAFKA_ENV))
+      .kafkaHost(config.getString(KAFKA_HOST))
+      .kafkaPort(config.getString(KAFKA_PORT))
+      .okapiUrl(config.getString(OKAPI_URL))
+      .replicationFactor(Integer.parseInt(config.getString(KAFKA_REPLICATION_FACTOR)))
+      .maxRequestSize(Integer.parseInt(config.getString(KAFKA_MAX_REQUEST_SIZE)))
+      .build();
+    LOGGER.info(format("kafkaConfig: %s", kafkaConfig));
+    EventManager.registerKafkaEventPublisher(kafkaConfig, vertx, maxDistributionNumber);
```

Comment:
With my limited knowledge of the app. It seems the line would be called somewhere else already. There is nothing unique about the kafkaConfig class here.